### ISSUE TITLE
Unable to fetch chat history

### ISSUE THREAD

**Issue by PonnappanG** [descentis/lab-code#1]
Getting answer for the first question. but second question throwing this error.

ValueError: Unsupported chat history format: <class 'str'>. Full chat history: Human: tell me about the candidate AI: Based on the provided text, here is what can be inferred about the candidate: * The candidate has experience in quality control, inventory management, and process improvement, as well as collaboration with developers.

Traceback:
File "C:\Users\Ponnappan G\AppData\Local\Programs\Python\Python310\lib\site-packages\streamlit\runtime\scriptrunner\exec_code.py", line 88, in exec_func_with_error_handling
    result = func()
File "C:\Users\Ponnappan G\AppData\Local\Programs\Python\Python310\lib\site-packages\streamlit\runtime\scriptrunner\script_runner.py", line 579, in code_to_exec
    exec(code, module.__dict__)
File "E:\Ponnappan G\GenAI\lab-code\PonnappanG\Streamlit\app.py", line 84, in <module>
    response = st.session_state.conversation_chain({'question': user_input})
File "C:\Users\Ponnappan G\AppData\Local\Programs\Python\Python310\lib\site-packages\langchain_core\_api\deprecation.py", line 182, in warning_emitting_wrapper
    return wrapped(*args, **kwargs)
File "C:\Users\Ponnappan G\AppData\Local\Programs\Python\Python310\lib\site-packages\langchain\chains\base.py", line 389, in __call__
    return self.invoke(
File "C:\Users\Ponnappan G\AppData\Local\Programs\Python\Python310\lib\site-packages\langchain\chains\base.py", line 170, in invoke
    raise e
File "C:\Users\Ponnappan G\AppData\Local\Programs\Python\Python310\lib\site-packages\langchain\chains\base.py", line 160, in invoke
    self._call(inputs, run_manager=run_manager)
File "C:\Users\Ponnappan G\AppData\Local\Programs\Python\Python310\lib\site-packages\langchain\chains\conversational_retrieval\base.py", line 146, in _call
    chat_history_str = get_chat_history(inputs["chat_history"])
File "C:\Users\Ponnappan G\AppData\Local\Programs\Python\Python310\lib\site-packages\langchain\chains\conversational_retrieval\base.py", line 55, in _get_chat_history
    raise ValueError(

**Comment by PonnappanG** [comment]
PonnappanG/app.py - file

**Comment by descentis** [comment]
Can you please post the whole error message?

**Comment by PonnappanG** [comment]
Human: tell me about the candidate


> Finished chain.


> Finished chain.

────────────────────── Traceback (most recent call last) ───────────────────────

  /home/adminuser/venv/lib/python3.10/site-packages/streamlit/runtime/scriptru  

  nner/exec_code.py:88 in exec_func_with_error_handling                         

                                                                                

  /home/adminuser/venv/lib/python3.10/site-packages/streamlit/runtime/scriptru  

  nner/script_runner.py:579 in code_to_exec                                     

                                                                                

  /mount/src/ponnappang/Streamlit/app.py:86 in <module>                         

                                                                                

    83 │   │   st.markdown(user_input)                                          

    84 │                                                                        

    85 │   with st.chat_message("assistant"):                                   

  ❱ 86 │   │   response = st.session_state.conversation_chain({'question': use  

    87 │   │   assistant_response = response["answer"]                          

    88 │   │   st.markdown(assistant_response)                                  

    89 │   st.session_state.chat_history.append({'role': "assistant", "content  

                                                                                

  /home/adminuser/venv/lib/python3.10/site-packages/langchain_core/_api/deprec  

  ation.py:182 in warning_emitting_wrapper                                      

                                                                                

    179 │   │   │   if not warned and not is_caller_internal():                 

    180 │   │   │   │   warned = True                                           

    181 │   │   │   │   emit_warning()                                          

  ❱ 182 │   │   │   return wrapped(*args, **kwargs)                             

    183 │   │                                                                   

    184 │   │   async def awarning_emitting_wrapper(*args: Any, **kwargs: Any)  

    185 │   │   │   """Same as warning_emitting_wrapper, but for async functio  

                                                                                

  /home/adminuser/venv/lib/python3.10/site-packages/langchain/chains/base.py:3  

  89 in __call__                                                                

                                                                                

    386 │   │   │   "run_name": run_name,                                       

    387 │   │   }                                                               

    388 │   │                                                                   

  ❱ 389 │   │   return self.invoke(                                             

    390 │   │   │   inputs,                                                     

    391 │   │   │   cast(RunnableConfig, {k: v for k, v in config.items() if v  

    392 │   │   │   return_only_outputs=return_only_outputs,                    

                                                                                

  /home/adminuser/venv/lib/python3.10/site-packages/langchain/chains/base.py:1  

  70 in invoke                                                                  

                                                                                

    167 │   │   │   )                                                           

    168 │   │   except BaseException as e:                                      

    169 │   │   │   run_manager.on_chain_error(e)                               

  ❱ 170 │   │   │   raise e                                                     

    171 │   │   run_manager.on_chain_end(outputs)                               

    172 │   │                                                                   

    173 │   │   if include_run_info:                                            

                                                                                

  /home/adminuser/venv/lib/python3.10/site-packages/langchain/chains/base.py:1  

  60 in invoke                                                                  

                                                                                

    157 │   │   try:                                                            

    158 │   │   │   self._validate_inputs(inputs)                               

    159 │   │   │   outputs = (                                                 

  ❱ 160 │   │   │   │   self._call(inputs, run_manager=run_manager)             

    161 │   │   │   │   if new_arg_supported                                    

    162 │   │   │   │   else self._call(inputs)                                 

    163 │   │   │   )                                                           

                                                                                

  /home/adminuser/venv/lib/python3.10/site-packages/langchain/chains/conversat  

  ional_retrieval/base.py:146 in _call                                          

                                                                                

    143 │   │   _run_manager = run_manager or CallbackManagerForChainRun.get_n  

    144 │   │   question = inputs["question"]                                   

    145 │   │   get_chat_history = self.get_chat_history or _get_chat_history   

  ❱ 146 │   │   chat_history_str = get_chat_history(inputs["chat_history"])     

    147 │   │                                                                   

    148 │   │   if chat_history_str:                                            

    149 │   │   │   callbacks = _run_manager.get_child()                        

                                                                                

  /home/adminuser/venv/lib/python3.10/site-packages/langchain/chains/conversat  

  ional_retrieval/base.py:55 in _get_chat_history                               

                                                                                

     52 │   │   │   ai = "Assistant: " + dialogue_turn[1]                       

     53 │   │   │   buffer += "\n" + "\n".join([human, ai])                     

     54 │   │   else:                                                           

  ❱  55 │   │   │   raise ValueError(                                           

     56 │   │   │   │   f"Unsupported chat history format: {type(dialogue_turn  

     57 │   │   │   │   f" Full chat history: {chat_history} "                  

     58 │   │   │   )                                                           

────────────────────────────────────────────────────────────────────────────────

ValueError: Unsupported chat history format: <class 'str'>. Full chat history: 

Human: tell me about the candidate

AI: Based on the provided text, here is what can be inferred about the 

candidate:

**Comment by PonnappanG** [comment]
![image](https://github.com/user-attachments/assets/2b985b3c-6a9e-4f5b-a076-3ae869ea9ded)

**Comment by PonnappanG** [comment]
it seems on conversation_chain

**Comment by descentis** [comment]
Is this happening for all different inputs?

**Comment by PonnappanG** [comment]
Yes. Also, I mimic the testBot code from other repository. Getting same issue. Seems it is because of requirement modules/python packages.

**Comment by descentis** [comment]
This is weird, I tried mimicking your error but couldn't reproduce it. Can you tell me about the python version that you are using?

**Comment by PonnappanG** [comment]
3.10

**Comment by descentis** [comment]
Were you able to resolve this?

**Comment by PonnappanG** [comment]
Hi Amit,
It is working now. 
It is because of Indentation of that section. I faced the issue while I did the labs. So applied the same here. I clear the indentations and again provide the indent until the app stop throwing error. After that its working good.
![image](https://github.com/user-attachments/assets/1ecc1e64-421b-4322-bb38-d81fa29877b0)


Thanks. 

Anyway it went upto my linked in profile from that document.. That is weird!!!!
